tests/rest/admin/test_user.py

```diff
@@ -516,6 +570,103 @@ def test_create_user(self):
         self.assertEqual(False, channel.json_body["is_guest"])
         self.assertEqual(False, channel.json_body["deactivated"])
 
+    def test_create_user_mau_limit_reached_active_admin(self):
+        """
+        Check that a new regular user is created successfully if MAU limit is reached.
+        Admin user was active before creating user.
+        """
+        self.hs.config.registration_shared_secret = None
+
+        handler = self.hs.get_registration_handler()
+        auth_blocking = self.hs.get_auth()._auth_blocking
+
+        # Configure MAU limit
+        auth_blocking._limit_usage_by_mau = True
+        auth_blocking._max_mau_value = 2
+        auth_blocking._mau_trial_days = 0
+
+        # Sync to set admin user to active
+        # before limit of monthly active users is reached
+        request, channel = self.make_request(
+            "GET", "/sync", access_token=self.admin_user_tok
+        )
+        self.render(request)
+
+        if channel.code != 200:
+            raise HttpResponseException(
+                channel.code, channel.result["reason"], channel.result["body"]
+            ).to_synapse_error()
+
+        # Set monthly active users to the limit
+        self.store.get_monthly_active_count = Mock(
+            return_value=defer.succeed(auth_blocking._max_mau_value)
+        )
+        # The registration of a new user fails due to the limit
+        self.get_failure(
+            handler.register_user(localpart="local_part"), ResourceLimitError
+        )
+
+        # Register new user with admin API
+        url = "/_synapse/admin/v2/users/@bob:test"
+
+        # Create user
+        body = json.dumps({"password": "abc123", "admin": False})
+
+        request, channel = self.make_request(
+            "PUT",
+            url,
+            access_token=self.admin_user_tok,
+            content=body.encode(encoding="utf_8"),
+        )
+        self.render(request)
+
+        self.assertEqual(201, int(channel.result["code"]), msg=channel.result["body"])
+        self.assertEqual("@bob:test", channel.json_body["name"])
+        self.assertEqual(False, channel.json_body["admin"])
+
+    def test_create_user_mau_limit_reached_passiv_admin(self):
+        """
+        Try to create a new regular user if MAU limit is reached.
+        Admin user was not active before creating user.
```

Comment:
I'm still a bit baffled by why we need this as well as the previous test. Is it really likely that an inactive admin could be blocked from creating a new user?

Reply:
The test is presented for completeness. It can be left out. We deal with mau. And it would be possible that an active and passive admin/user get different results.